docs/PID.md

```diff
@@ -0,0 +1,109 @@
+temperature, it will produce biased and asymmetric results. As the plot above
+shows, the system spends more time with the power off, and has a greater
+amplitude above the target temperature than it does below.
+
+In an ideal system, both power on and off time, as well as the amplitude above
+and below the target temperature would be equal. 3d printers don’t actively
+cool the hot end or bed, so they can never reach the ideal.
+
+The following plot is a relay test that uses the methodology laid out by
+Taysom & Sorensen. After each iteration, the data is analyzed, and a new
+maximum power setting is calculated. As can bee seen from the plot, the system
+starts the test asymmetric, but ends it very symmetric.
+
+![advanced relay test](img/pid_02.png)
+
+ A topic that’s not often discussed in the 3dp community is the conditions in
+ which calibration should be performed. When a calibration test is performed
+ external variables should be minimized as much as possible, as the goal of the
+test is to model the system in steady state conditions free of external
+disturbances. For example, if you are calibrating a hot end, you do not want a
+bed or chamber heater actively heating up or cooling down. You want them off,
+or holding at their target temperature. Part cooling and chamber fans can also
+be problematic, as they can cause temperature fluctuations in the hot end.
+
+### Pid Control Parameters
+
+Many methods exist for calculating control parameters, such as Ziegler-Nichols,
+Cohen-Coon, Kappa-Tau, Lambda, and many more. By default, classical
+Ziegler-Nichols parameters are generated. If A user wants to experiment with
+other flavors of Ziegler-Nichols, or Cohen-Coon parameters, they can extract
+the constants from the log as seen below and feed them into this
+[spreadsheet](resources/pid_params.xls).
+
+```
+Ziegler-Nichols constants: Ku=0.103092 Tu=41.800000
+Cohen-Coon constants: Km=-17.734845 Theta=6.600000 Tau=-10.182680
+```
+
+Classical Ziegler-Nichols parameters work in all scenarios. Cohen-Coon
+parameters work better with systems with a large amount of dead time/delay.
+For example, if a printer has a large thermal mass bed that’s slow to heat up
+and stabilize, Cohen-Coon parameters will generally do a better job at
+controlling it.
+
+### Pid Control Algorithms
+
+Klipper currently supports two control algorithems positional, and velocity.
+The fundamental difference between the two algorithms is as follows. The
+positional algorithm calculates what the PWM value should be for the current
+time interval. The velocity algorithm calculates how much the previous PWM
+setting should be changed by, to get the PWM value for the current time
+interval.
+
+Positional is the default algorithm, as it will work in every scenario. The
+Velocity algorithm can provide superior results to the Positional algorithm,
+but requires lower noise sensor readings, or a larger smoothing time setting.
```

Comment:
```suggestion
Positional is the default algorithm, as it will work in every scenario. The
Velocity algorithm can provide superior results to the Positional algorithm but
requires lower noise sensor readings, or a larger smoothing time setting.
```